gnucash/report/trep-engine.scm

```diff
@@ -154,7 +162,51 @@ in the Options panel."))
         (cons #\f (G_ "Frozen"))
         (cons #\v (G_ "Voided"))))
 
-(define (sortkey-list split-action?)
+;; Hashtable to memorize both the sortvalue and rendering of the
+;; split's matching hashtag 
+(define hashtag-htable (make-hash-table))
```

Comment:
this `hashtag-htable` is a global var *within* trep-engine. It means repeated calls to this trep-engine will fail to refresh the hash table. Either reset the hash table on launching trep-renderer, or (better) create a split->hashtag generator similarly to `gnc:make-split->owner` and the generator will automatically clean up when gc runs.

Reply:
I didn't realize anything survives pass a report run. I was (wrongly) under the impression everything gets cleaned up. So yes that won't work. I assumed (wrongly) your reference to `gnc:make-split->owner` was for the hashtable part, not the other stuff. In `gnc:make-split->owner` I do not understand how the hashtable created within `gnc:make-split->owner` remains initialized through what I assume must be repeated calls to `gnc:make-split->owner`. In fact in a first version I put mine in there as well and of course I even got worse performance. So I am clearly missing the part about how this "generator" is initialized somehow. Possibly the answer is in your candidate below, so if it is let me look in there.

Reply:
> I didn't realize anything survives pass a report run. I was (wrongly) under the impression everything gets cleaned up. So yes that won't work. I assumed (wrongly) your reference to `gnc:make-split->owner` was for the hashtable part, not the other stuff. In `gnc:make-split->owner` I do not understand how the hashtable created within `gnc:make-split->owner` remains initialized through what I assume must be repeated calls to `gnc:make-split->owner`. In fact in a first version I put mine in there as well and of course I even got worse performance. So I am clearly missing the part about how this "generator" is initialized somehow. Possibly the answer is in your candidate below, so if it is let me look in there.

`gnc:make-split->owner` does not return the split's owner. It _creates_ a function (which is aptly named split->owner) which takes a split and returns its owner, and also maintains its own hash table. When the split->owner is out of scope, it is garbage collected as well as the hash table.

Reply:
OK I am going to spend more time looking at that `gnc:make-split->owner` if only to learn how it's doing this.